docs/admin/etcd.md

```diff
@@ -1,49 +1,163 @@
 ---
 assignees:
-- lavalamp
-title: Configuring Kubernetes Use of etcd
+- mml
+- wojtek-t
+title: Upgrading and Rolling Back etcd
 ---
 
+## About etcd
+
 [etcd](https://coreos.com/etcd/docs/latest/) is a highly-available key value
 store which Kubernetes uses for persistent storage of all of its REST API
-objects.
+objects.  For the mechanics behind how kubernetes builds, distributes and
+deploys Kubernetes, see _some doc_.
+
+## Background
+
+As of 1.5.1 kubernetes release, we are still using etcd from 2.2.1 release with
+v2 API.  Also, we have no pre-existing process for updating etcd, as we have
+never updated etcd by either minor or major version.
+
+Note that we need to migrate both: etcd versions that we are using (from 2.2.1
+to at least 3.0.x) as well as the version of API Kubernetes talks to etcd (3.0.x
+etcd binaries support both v2 and v3 API).
+
+This document describes how to do this migration.
+
+### Etcd upgrade limitations
+
+The way etcd was designed introduces some significant limitations on how the
+upgrade can be done. These are the main limitations. 
+
+#### One minor release at a time
+
+It is possible to upgrade only by one minor release at a time (though, within
+patch releases you can switch versions at any time back and forth). That
+basically means, that we cannot upgrade directly e.g. from 2.1.x to 2.3.x.
+
+Fortunately, this limitation it is easy to workaround - it is enough to start a
+cluster for any intermediate minor release, wait until it is healthy and
+functional and stop it then. This will do the migration itself underneath. 
+
+As an example: to upgrade from 2.1.x to 2.3.y version, it is enough to start
+etcd in 2.2.z version, wait until it is healthy, stop it and then start then
+2.3.y version, wait until it is up and we are done.
+
+#### No rollback
+
+Etcd doesn’t support rollback procedure at all (by design). That means, that if
+you migrate your cluster e.g. from 2.2.x to 2.3.y, there is no way to get back
+to 2.2.x (other than restoring from backup data from the moment when 2.2.x was
+running, though if we were running 2.3.y for some time, we will lose all data
+that were written in the meantime).
```

Comment:
may be gramatically less controversial to say "all data written in the meantime"